### ISSUE TITLE
ibv_devinfo doesnt show link layer info

### ISSUE THREAD

**Issue by thamizh1990** [SoftRoCE/rxe-dev#46]
root@gl8:/home/gl8/Desktop/nvme_softroce/librxe-dev# ibv_devinfo 
hca_id: rxe0
    transport:          InfiniBand (0)
    fw_ver:             0.0.0
    node_guid:          428d:5cff:fe35:bb12
    sys_image_guid:         0000:0000:0000:0000
    vendor_id:          0x0000
    vendor_part_id:         0
    hw_ver:             0x0
    phys_port_cnt:          1
        port:   1
            state:          PORT_ACTIVE (4)
            max_mtu:        4096 (5)
            active_mtu:     1024 (3)
            sm_lid:         0
            port_lid:       0
            port_lmc:       0x00

**Comment by psebexen** [comment]
Are there any additional symptoms related to the rxe0 device, or is this issue only about the ibv_devinfo output? Running `ibv_devinfo -v` should print more information or you can manually inspect the rxe0 device in `/sys/class/infiniband/rxe0`.

**Comment by thamizh1990** [comment]
thanks for the solution, i have found the link_layer info in /sys/class/infiniband/rxe0 , 
i need some clearance , actually my environment setup is " two pc connected with ethernet cable" , can i able to test this SoftRoCE , or i need any other special fabrics interface

**Comment by psebexen** [comment]
Yes; your environment sounds sufficient. If the computers can send and receive normal IP traffic over Ethernet, they should be able to run the SoftRoCE driver. There are a number of user space packages, as listed in the wiki, that are helpful for using and configuring the connection, but the basic functionality is provided by code in the kernel source tree. 
If you are interested in how this works, I recommend looking at the end of the file `drivers/infiniband/hw/rxe/rxe_net.c` in this repository as a starting point - a UDP socket is created and some of the net device settings are configured there at initialization time.
